### PR TITLE
Various emitter cleanup

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Added support for unsigned int types.
Removed some commented out and dead code. This includes the commented out code for client-side default values in models as tsp doesn't support that scenario.
Consolidated some type adaptation.
Removed some now-handled TODOs.
No more unused locals.

### CODE REVIEW

File: packages/typespec-go/src/tcgcadapter/types.ts [hunk 1]
```diff
@@ -677,16 +592,27 @@ export class typeAdapter {
...
+      }
+      case 'decimal':
+      case 'decimal128': {
+        const keyName = `literal-${constType.valueType.kind}-${constType.value}`;
+        let literalDecimal = this.types.get(keyName);
+        if (literalDecimal) {
+          return <go.LiteralValue>literalDecimal;
+        }
+        literalDecimal = new go.LiteralValue(new go.PrimitiveType('float64'), constType.value);
+        this. types.set(keyName, literalDecimal);
```
Comment (chlowell) [inline packages/typespec-go/src/tcgcadapter/types.ts:613]:
```suggestion
        this.types.set(keyName, literalDecimal);
```

**Review by chlowell — APPROVED** [review]

File: packages/typespec-go/src/emitter.ts [hunk 2]
```diff
@@ -15,22 +15,13 @@ import { generatePolymorphicHelpers } from '../../codegen.go/src/polymorphics.js
 import { generateResponses } from '../../codegen.go/src/responses.js';
 import { generateTimeHelpers } from '../../codegen.go/src/time.js';
 import { generateServers } from '../../codegen.go/src/fake/servers.js';
-import { generateServerFactory } from '../../codegen.go/src/fake/factory.js';
+//import { generateServerFactory } from '../../codegen.go/src/fake/factory.js';
```
Comment (RickWinter) [inline packages/typespec-go/src/emitter.ts:18]: Remove the commented code

File: packages/typespec-go/src/tcgcadapter/types.ts [hunk 3]
```diff
@@ -161,15 +161,6 @@ export class typeAdapter {
         types.set(m4.SchemaType.Binary, binaryType);
         return binaryType;
       }*/
```
Comment (RickWinter) [inline packages/typespec-go/src/tcgcadapter/types.ts:163]: Is there a reason to leave the commented code above? Is that a tcgc bug that when fixed we will enable?

File: packages/typespec-go/src/tcgcadapter/types.ts [hunk 4]
```diff
@@ -331,51 +322,32 @@ export class typeAdapter {
...
-      case 'int64': {
-        const int64Key = 'int64';
-        let int64 = this.types.get(int64Key);
-        if (int64) {
-          return int64;
-        }
-        int64 = new go.PrimitiveType(int64Key);
-        this.types.set(int64Key, int64);
-        return int64;
+      case 'int8':
```
Comment (RickWinter) [inline packages/typespec-go/src/tcgcadapter/types.ts:328]: Nice consolidation :)

File: packages/typespec-go/src/tcgcadapter/types.ts [hunk 5]
```diff
@@ -572,43 +523,7 @@ export class typeAdapter {
...
-          literalValue = new go.LiteralValue(field.type, prop.clientDefaultValue);
-          types.set(keyName, literalValue);
-        }
-        field.defaultValue = <go.LiteralValue>literalValue;
-      }
-    }*/
+    }
   
     // TODO: XMLInfo
     //field.xml = adaptXMLInfo(prop.schema);
```
Comment (RickWinter) [inline packages/typespec-go/src/tcgcadapter/types.ts:529]: Should this TODO remain?

**Review by RickWinter — APPROVED** [review]

File: packages/typespec-go/src/tcgcadapter/types.ts (hunk 3 above)
Comment (jhendrixMSFT) [reply packages/typespec-go/src/tcgcadapter/types.ts:163]: Streaming content still needs to be supported. It's WIP with the changes for multipart/form data. Will remove with those changes.

File: packages/typespec-go/src/emitter.ts (hunk 2 above)
Comment (jhendrixMSFT) [reply packages/typespec-go/src/emitter.ts:18]: I left this and the point of invocation as we still need to decide what we'll do with the fake server factory (will update comment with TODO).

File: packages/typespec-go/src/tcgcadapter/types.ts (hunk 5 above)
Comment (jhendrixMSFT) [reply packages/typespec-go/src/tcgcadapter/types.ts:529]: XML support hasn't been wired up yet and I don't know if tsp and/or tcgc even support it (or if they ever will).

File: packages/typespec-go/src/tcgcadapter/types.ts (hunk 1 above)
Comment (jhendrixMSFT) [reply packages/typespec-go/src/tcgcadapter/types.ts]: Fixed (also found another instance).

**Pull request merged**